File: app/src/main/java/com/rsupport/mobile1/test/utils/manager/NetworkManager.java
package com.rsupport.mobile1.test.utils.manager;


import android.app.Activity;
import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.os.Build;

public class NetworkManager {
    public final static String LTE = "LTE";
    public final static String WIFI = "WIFI";
    public static String method;
    public static boolean isConnect;

    public static boolean isConnect(Context context) {
        //롤리팝 이상
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            return isConnect;
        }
        //미만 직접 찾기
        else {
            getNetworkStatus(context);
            return isConnect;
        }
    }

    @Override
    public String toString() {
        return " method isConnect" + isConnect;
    }

    public static void getNetworkStatus(Context context) {
        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        boolean isMetered = cm.isActiveNetworkMetered();
        if (isMetered) {
            method = LTE;
        } else {
            method = WIFI;
        }

        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        isConnect = activeNetwork != null && activeNetwork.isConnectedOrConnecting();
        if (!isConnect) {
            method = "";
        }
    }

}

File: app/src/main/java/com/rsupport/mobile1/test/utils/manager/NetworkManager.kt
package com.rsupport.mobile1.test.utils.manager

import android.content.Context
import com.rsupport.mobile1.test.utils.manager.NetworkManager
import android.os.Build
import android.net.ConnectivityManager
import android.net.NetworkInfo

class NetworkManager {
    override fun toString(): String {
        return " method isConnect" + isConnect
    }

    companion object {
        const val LTE = "LTE"
        const val WIFI = "WIFI"
        var method: String? = null
        var isConnect = false
        @JvmStatic
        fun isConnect(context: Context): Boolean {
            //롤리팝 이상 콜백으로 네트워크 연결상태 설정
            //롤리팝 미만 네트워크 연결상태 확인
            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
                getNetworkStatus(context)
            }
            return isConnect
        }

        //네트워크 연결상태 설정
        @JvmStatic
        fun getNetworkStatus(context: Context) {
            val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val isMetered = cm.isActiveNetworkMetered
            if (isMetered) {
                method = LTE
            } else {
                method = WIFI
            }
            val activeNetwork = cm.activeNetworkInfo
            isConnect = activeNetwork != null && activeNetwork.isConnectedOrConnecting
            if (!isConnect) {
                method = ""
            }
        }
    }
}